WordPress/src/main/java/org/wordpress/android/ui/posts/UpdatePostCategoriesUseCase.kt

```diff
@@ -1,14 +1,26 @@
 package org.wordpress.android.ui.posts
 
 import dagger.Reusable
+import org.wordpress.android.analytics.AnalyticsTracker.Stat
+import org.wordpress.android.fluxc.model.PostImmutableModel
+import org.wordpress.android.ui.posts.EditPostRepository.UpdatePostResult
+import org.wordpress.android.ui.posts.EditPostRepository.UpdatePostResult.Updated
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
 import javax.inject.Inject
 
 @Reusable
-class UpdatePostCategoriesUseCase @Inject constructor() {
+class UpdatePostCategoriesUseCase @Inject constructor(
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper
+) {
     fun updateCategories(categoryList: List<Long>, editPostRepository: EditPostRepository) {
         editPostRepository.updateAsync({ postModel ->
             postModel.setCategoryIdList(categoryList)
             true
+        }, { postModel: PostImmutableModel?, result: UpdatePostResult ->
+            if (result == Updated) {
+                analyticsTrackerWrapper.trackPrepublishingNudges(Stat.EDITOR_POST_CATEGORIES_ADDED)
```

Comment:
This tracking makes this usecase not-reusable, wdyt?

Reply:
> This tracking makes this usecase not-reusable, wdyt?
You are correct, but it was the lesser of two evils. The save happens on <- back button press and no one is listening for success/failure. Now that I am no longer rushing for this release, I have some time to figure out how to get the notification from the closed view to the previous.

Reply:
I'd consider inlining the usecase. It doesn't do much anyway, wdyt?

Reply:
Easier to mock the useCase for testing, but let me check out inlining and see what happens. Thanks for the suggestion.

Reply:
Moved inline in 0d17dc5 👍